source/common/filesystem/inotify/watcher_impl.cc

```diff
@@ -54,7 +54,7 @@ absl::Status WatcherImpl::addWatch(absl::string_view path, uint32_t events, OnCh
 
 absl::Status WatcherImpl::onInotifyEvent() {
   while (true) {
-    uint8_t buffer[sizeof(inotify_event) + NAME_MAX + 1];
+    alignas(inotify_event) uint8_t buffer[sizeof(inotify_event) + NAME_MAX + 1];
```

Comment:
If I'm understanding this code correctly (which I might not be) I'm not sure this is safe enough. The SIGTRAP happens, I believe, when we cast `&buffer[index]` to `inotify_event*` and then try to use that object. Is that right? If so, I think this change here solves the problem for the first event the buffer, but not necessarily any subsequent ones. Down on line 98, we do `index += sizeof(inotify_event) + file_event->len;` which,  I think, means that subsequent iterations through the loop are not guaranteed to align. Instead, I think it might make sense to replace the cast with:

```
inotify_event* file_event;
safeMemcpyUnsafeSrc(&file_event, &buffer[index]);
````

which ensures that `file_event` is properly aligned, regardless of how `buffer` is laid out. What do you think?

Reply:
Well spotted. However per the doc the the kernel will align subsequent inotify_event structures if it returns more then one. So the caller is only responsible for aligning the first one.

Reply:
Oh, fascinating! Do you have a pointer to the docs that we could link to here for future readers?

Reply:
Oh, from the man page:
```
       The name field is present only when an event is returned for a
       file inside a watched directory; it identifies the filename
       within the watched directory.  This filename is null-terminated,
       and may include further null bytes ('\0') to align subsequent
       reads to a suitable address boundary.
```
Mind pasting this in (or summarizing it)?

Reply:
Added comment

Reply:
TIL.
Thanks!

Reply:
Great, thanks!